code/modules/organs/subtypes/parasite.dm

```diff
@@ -0,0 +1,176 @@
+/obj/item/organ/parasite
+	name = "parasite"
+	icon = 'icons/mob/alien.dmi'
+	icon_state = "burst_lie"
+	dead_icon = "bursted_lie"
+
+	organ_tag = "parasite"
+	var/stage = 1
+	var/max_stage = 4
+	var/stage_ticker = 0
+	var/stage_interval = 600 //time between stages, in seconds
+
+/obj/item/organ/parasite/process()
+	..()
+
+	if(!owner)
+		return
+
+	if(stage < max_stage)
+		stage_ticker += 2 //process ticks every ~2 seconds
+
+	if(stage_ticker >= stage*stage_interval)
+		stage = min(stage+1,max_stage)
+
+///////////////////
+///K'ois Mycosis///
+///////////////////
+
+/obj/item/organ/parasite/kois
+	name = "k'ois mycosis"
+	//icon_state = "kois-on"
+	//dead_icon = "kois-off"
+
+	organ_tag = "kois"
+
+	parent_organ = "chest"
+
+/obj/item/organ/parasite/kois/process()
+	..()
+
+	if(stage >= 1)
+		if(prob(5*stage))
+			owner << "<span class='warning'>You feel a stinging pain in your abdomen!</span>"
+			owner.emote("me",1,"winces slightly.")
+			owner.adjustHalLoss(5)
+
+		if(prob(10))
+			owner.emote("cough")
+
+		if(prob(5))
+			spawn owner.emote("me", 1, "coughs up blood!")
+			owner.drip(10)
+
+
+	if(stage >= 2 && prob(10))
+		spawn owner.emote("me", 1, "gasps for air!")
+		owner.losebreath += 15
+
+	if(stage >= 3)
+		set_light(1, l_color = "#E6E600")
+		if(prob(10))
+			owner << "<span class='warning'>You feel something squirming inside of you!</span>"
+			owner.reagents.add_reagent("phoron", 8)
+			owner.reagents.add_reagent("koispaste", 5)
+
+	if(stage >= 4 && prob(15))
+		owner << "<span class='danger'>You feel something alien coming up your throat!</span>"
+		owner.emote("cough")
+
+		var/turf/T = get_turf(owner)
+
+		var/datum/reagents/R = new/datum/reagents(100)
+		R.add_reagent("koispaste",10)
+		R.add_reagent("phoron",10)
+		var/datum/effect/effect/system/smoke_spread/chem/spores/S = new("koisspore")
+
+		S.attach(T)
+		S.set_up(R, 15, 0, T, 40)
+		S.start()
+
+		owner.adjustHalLoss(15)
+		owner.drip(15)
+		owner.delayed_vomit()
+
+///////////////////
+///Black Mycosis///
+///////////////////
+
+/obj/item/organ/parasite/blackkois
+	name = "k'ois mycosis"
+	//icon_state = "blackkois-on"
+	//dead_icon = "blackkois-off"
+
+	organ_tag = "blackkois"
+
+	parent_organ = "head"
+
+/obj/item/organ/parasite/blackkois/process()
+	..()
+
+	if(stage >= 1 && prob(5*stage))
+		if(stage < 3)
+			owner << "<span class='warning'>You feel a stinging pain in your abdomen!</span>"
+		else
+			owner << "<span class='warning'>You feel a stinging pain in your head!</span>"
+		owner.emote("me",1,"winces slightly.")
+		owner.adjustHalLoss(5)
+
+	if(stage >= 2)
+		if(stage < 3 && prob(10))
+			spawn owner.emote("me", 1, "gasps for air!")
+			owner.losebreath += 15
+
+		else if(prob(5*stage))
+			owner << "<span class='warning'>You feel disorientated!</span>"
+			switch(rand(1,3))
+				if(1)
+					owner.confused += 10
+					owner.apply_effect(10,EYE_BLUR)
+				if(2)
+					owner.slurring += 30
+				if(3)
+					owner.dizziness += 10
```

Comment:
Isn't this supposed to be done via. proc?

Reply:
I don't think so?